### PR TITLE
Add digest worker with Slack service powered by Oban

### PR DESCRIPTION
**Pull request by simonprev**
We add a `TelemetryUI.Digest.Worker` that can be scheduled through Oban to send messages to Slack.
The goal is to expose the metrics instead of relying on the user to go to the metrics page each week.

The share links use absolute time meaning that we can open an old shared link to compare metrics between 2 time frames.

**We can also configure some parameters:** 

- The time frame we want to share (Last 7 days, last month, last 2 hours, etc)
- The pages we want to expose
- The TelemetryUI instance we want to share (we could start an instance of TelemetryUI in application.ex that we don’™ expose in the router and only use in the digest)
- The Slack message (icon, header, username)

### CODE REVIEW

File: guides/slack-digest.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+`telemetry_ui` has amodule to send shareable links to external resources. The only resource implemented for now is Slack.
+
+To schedule a weekly digest, we rely on [Oban](https://hexdocs.pm/oban/) to have a safe, unique and _schedulable_ queue across our system.
+
+**Requirements:**
+You will need to [install it before continuing this guide](https://hexdocs.pm/oban/installation.html).
+You will also need to have a valid [Slack hook URL](https://api.slack.com/messaging/webhooks#enable_webhooks).
+
+---
```
Comment (remi) [inline guides/slack-digest.md]:
```suggestion
## Requirements

You will need to [install it](https://hexdocs.pm/oban/installation.html) before continuing this guide.

You will also need to have a valid [Slack hook URL](https://api.slack.com/messaging/webhooks#enable_webhooks).

## Usage
```

File: guides/slack-digest.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+     crontab: [
+       {"@weekly", TelemetryUI.Digest.Worker, args: telemetry_ui_digest_args}
+     ]}
+  ],
+  queues: [default: 10]
+```
+
+Here is the kind of result you should see in Slack:
+
+<img style="border-radius: 4px" width="321" alt="image" src="https://user-images.githubusercontent.com/464900/204909541-89df8db6-348e-4134-81f9-8bab89518dff.png">
```
Comment (remi) [inline guides/slack-digest.md]: I think the markdown shouldn’t include `style` attributes.

```suggestion
<img width="321" src="https://user-images.githubusercontent.com/464900/204909541-89df8db6-348e-4134-81f9-8bab89518dff.png">
```

Comment (remi) [inline guides/slack-digest.md]: Also, I think the image copy differs from the one in the code. There’s no “forever” in the code.

File: lib/telemetry_ui.ex [hunk 3]
```diff
@@ -102,7 +102,12 @@ defmodule TelemetryUI do
   def theme(name), do: config(name, :theme)
   def backend(name), do: config(name, :backend)
 
+  def valid_share_key?(share_key) do
+    is_binary(share_key) and byte_size(share_key) === 16
```
Comment (remi) [inline lib/telemetry_ui.ex]: Is `16` used elsewhere when we **generate** a share key? Should it not be duplicated here?

**Review by remi — APPROVED** [review]

File: lib/telemetry_ui.ex (hunk 3 above)
Comment (remi) [reply lib/telemetry_ui.ex]: I found what I was looking for:

* https://github.com/mirego/telemetry_ui/blob/master/lib/telemetry_ui/web/crypto.ex#L5
* https://github.com/mirego/telemetry_ui/blob/master/lib/telemetry_ui/web/crypto.ex#L8

So, shouldn’t all those `16` literals be stored as a single constant?

**Review by EtienneStPierre — APPROVED** [review]

**Pull request merged**